src/adcs_handler.c

```diff
@@ -46,242 +50,450 @@ ADCS_returnState adcs_telecommand(uint32_t * command, uint32_t length) {
   // uart_send(1, &ESC_CHAR);
   // uart_send(1, &EOM);
 
-  //get Acknowledgement 
+  // get Acknowledgement
   ADCS_returnState ack = ADCS_OK;
 
-  #ifdef USE_UART
-    ack = send_uart_telecommand(command, length);
-  #elif USE_I2C
-    ack = send_i2c_telecommand(command, length);
-  #endif
+#ifdef USE_UART
+  ack = send_uart_telecommand(command, length);
+#elif USE_I2C
+  ack = send_i2c_telecommand(command, length);
+#endif
 
   return ack;
 }
 
-/**
- * @brief
- * 		Perform a reset.
- * @details
- * 		<more detailed description of the function>
- * @return
- * 		<if applicable, brief description of what is returned by the function>
- */
-ADCS_returnState ADCS_reset() {
-  uint32_t command[2];
-  command[0] = RESET_ID;
-  command[1] = ADCS_MAGIC_NUMBER; // Magic number
-  return adcs_telecommand(command, 2);
-}
+ADCS_returnState adcs_telemetry(uint8_t TM_ID, uint8_t* reply,
+                                uint32_t length) {
+  ADCS_returnState ack = ADCS_OK;
+#ifdef USE_UART
+  ack = request_uart_telemetry(TM_ID, reply, length);
+#elif USE_I2C
+  ack = request_i2c_telemetry(TM_ID, reply, length);
+#endif
 
-/**
- * @brief
- * 		Reset pointer to log buffer (from where LastLogEvent TLM is returned)
- * @details
- * 		<more detailed description of the function>
- * 		<if applicable, brief description of what is returned by the function>
- */
-ADCS_returnState ADCS_reset_log_pointer() {
-  uint32_t command = RESET_LOG_POINTER_ID;
-  return adcs_telecommand(&command, 1);
+  return ack;
 }
 
-/**
- * @brief
- * 		Advance pointer to log buffer (from where LastLogEvent TLM is returned)
- * @details
- * 		<more detailed description of the function>
- * @return
- * 		<if applicable, brief description of what is returned by the function>
- */
-ADCS_returnState ADCS_advance_log_pointer() {
-  uint32_t command = ADVANCE_LOG_POINTER_ID;
-  return adcs_telecommand(&command, 1);
-}
+///**
```

Comment:
@vasu-gupta01 I just commented out your code for now because I changed the TC/TM functions and couldn't run tests with your code using the previous version. Not sure why GitHub is showing them like this.